src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -440,13 +440,13 @@ private DotnetHost DotnetHostInfo
 
                         if (ToolExe != _dotnetHostInfo.ToolNameOpt)
                         {
-                            _dotnetHostInfo = DotnetHost.CreateUnmanagedToolInvocation(ToolPath, commandLine);
+                            _dotnetHostInfo = DotnetHost.CreateUnmanagedToolInvocation(ToolExe, commandLine);
```

Comment:
How do we know this is an unmanaged tool? All we know is the name. 

Think we should consider "custom" vs "managed"

Reply:
If we're bikeshedding names, "custom" doesn't really make immediate sense (although it is better than "unmanaged"). The actual behavior we're trying to describe is "is invoked directly via the OS's process invocation utilities, as opposed to invoking it using any magic necessary to run .net IL (nothing on desktop)". "Custom" to me implies that it's a "user-specified executable", which may or may not be the case (in this case, it is indeed the case that they line up, but perhaps they might not in the future).

Maybe "native"?